### DISCUSSION TITLE
Cannot listen for Livewire events if component placed inside Filament widget

### DISCUSSION THREAD

**Discussion by nbyloff** [filamentphp/filament#11730] [Help] [answered]
### Package

Widgets

### Package Version

v3.2.44

### How can we help you?

I am trying to place a Livewire component inside a Filament Widget that has inline scripts. I have this inside my Livewire Component blade file:
````
@script
<script>
    console.log('started');
    document.addEventListener('livewire:init', function () {
        console.log('loaded');
    });
</script>
@endscript
````
It prints `started` but not `loaded`. However if I take the component and just render it on a page by itself without Filament, it does print `loaded`. Since Widgets build on top of Livewire components, I figured this would work out of the box, but can't get it to work.

EDIT: I will add that putting my code in an external file works and I've moved on with that, but prefer to do it inline so I am interested in how to make it work or if it should be working and it's not.

**Comment by leandrocfe** [comment]
add in the widget

```php
protected static bool $isLazy = false;
```

**Reply by nbyloff** [reply]
I just tried this, had no change.

**Reply by leandrocfe** [reply]
Did you clear the browser cache?

**Reply by nbyloff** [reply]
Yes, I even changed the console.log to make sure the updates were showing up. `XXX` prints, but not `YYY`.

livewire/active-call.blade.php
````
@script
<script>
    console.log('XXX');
    document.addEventListener('livewire:init', function () {
        console.log('YYY');
    });
</script>
@endscript
````


filament.widgets.active-call.blade.php
````
<x-filament-widgets::widget>

    <x-filament::section>
        <livewire:active-call />
    </x-filament::section>
</x-filament-widgets::widget>

````


````
<?php

namespace App\Filament\Widgets;

use App\Models\Customer;
use Filament\Widgets\Widget;
use Livewire\Attributes\On;

class ActiveCall extends Widget
{
    protected static string $view = 'filament.widgets.active-call';

    protected int | string | array $columnSpan = 'full';

    protected static ?int $sort = 1;

    protected static bool $isLazy = false;

}


````

**Reply by leandrocfe** [reply, marked as answer]
Try

```html
@push('scripts')
<script>
    console.log('XXX');
    document.addEventListener('livewire:init', function () {
        console.log('YYY');
    });
</script>
@endpush
```

**Reply by nbyloff** [reply]
That worked, thank you! Should I use `@push` for inline scripts going forward? I've only seen `@script` `@endscript` mentioned in the docs

**Reply by leandrocfe** [reply]
I'm not entirely sure if I can explain it clearly, but the '@push' directive is used to insert JavaScript into the HTML code, here:

https://github.com/filamentphp/filament/blob/3.x/packages/panels/resources/views/components/layout/base.blade.php
`@stack('scripts')`
